lib/OpenQA/Worker/Cache/Client.pm

```diff
@@ -0,0 +1,279 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Client;
+
+use OpenQA::Worker::Cache;
+use OpenQA::Worker::Common;
+
+use OpenQA::Worker::Cache::Request;
+use OpenQA::Worker::Cache::Request::Asset;
+use OpenQA::Worker::Cache::Request::Sync;
+
+use Mojo::Base 'Mojo::UserAgent';
+use Mojo::URL;
+use Mojo::File 'path';
+
+has host  => 'http://127.0.0.1:7844';
+has retry => 5;
+has cache_dir =>
+  sub { $ENV{CACHE_DIR} || (OpenQA::Worker::Common::read_worker_config(undef, undef))[0]->{CACHEDIRECTORY} };
+
+sub _url { Mojo::URL->new(shift->host)->path(shift)->to_string }
+
+sub _status {
+    return !!0 unless my $st = $_[0]->result->json->{status} // shift->result->json->{session_token};
+    return $st;
+}
+
+sub _result {
+    eval { $_[0]->post($_[0]->_url('status') => json => {lock => pop})->result->json->{+pop()} }
+}
+
+sub _query {
+    my ($self, $q) = @_;
+    _status(_retry(sub { $self->get($self->_url($q)) } => $self->retry));
+}
+
+sub _post {
+    my ($self, $q, $j) = @_;
+    _status(_retry(sub { $self->post($self->_url($q) => json => $j) } => $self->retry));
```

Comment:
The return value of this method seems significant, so it might be a good idea to make it obvious by adding a `return` keyword.